app/scripts/lib/stores/read-only-network-store.test.ts

```diff
@@ -0,0 +1,138 @@
+import log from 'loglevel';
+import nock from 'nock';
+import ReadOnlyNetworkStore from './read-only-network-store';
+import { MetaMaskStorageStructure } from './base-store';
+
+const FIXTURE_SERVER_HOST = 'localhost';
+const FIXTURE_SERVER_PORT = 12345;
+const FIXTURE_SERVER_ORIGIN = `http://${FIXTURE_SERVER_HOST}:${FIXTURE_SERVER_PORT}`;
+const FIXTURE_SERVER_PATH = '/state.json';
+
+const DEFAULT_INITIAL_STATE = {
+  data: { config: {} },
+};
+
+const MOCK_STATE = { data: { config: { foo: 'bar' } }, meta: { version: 1 } };
+
+/**
+ * Initiatilizes a ReadOnlyNetworkStore for testing
+ *
+ * @returns store - a ReadOnlyNetworkStore
+ */
+function setupReadOnlyNetworkStore() {
+  const store = new ReadOnlyNetworkStore();
+  return store;
+}
+
+/**
+ * Create a Nock scope for the fixture server response.
+ *
+ * @returns A Nock interceptor for the fixture server response.
+ */
+function mockFixtureServerInterceptor(): nock.Interceptor {
+  return nock(FIXTURE_SERVER_ORIGIN).get(FIXTURE_SERVER_PATH);
+}
+
+/**
+ * Create a Nock scope for the fixture server response, which will have a successful reply.
+ *
+ * @param state
+ */
+function setMockFixtureServerReply(
+  state: Record<string, unknown> = DEFAULT_INITIAL_STATE,
+): void {
+  mockFixtureServerInterceptor().reply(200, state);
+}
+
+describe('ReadOnlyNetworkStore', () => {
+  beforeEach(() => {
+    jest.resetModules();
+    nock.cleanAll();
+  });
+
+  describe('constructor', () => {
+    it('makes a request to the fixture server upon construction', async () => {
+      const scope = nock(FIXTURE_SERVER_ORIGIN)
+        .get(FIXTURE_SERVER_PATH)
+        .reply(200, 'Test');
+      expect(scope.pendingMocks().length).toBe(1);
+      setupReadOnlyNetworkStore();
+      await new Promise(process.nextTick);
+      expect(scope.pendingMocks().length).toBe(0);
+    });
+    it('loads state from the network if fetch is successful and response is ok', async () => {
+      setMockFixtureServerReply(MOCK_STATE);
+      const store = setupReadOnlyNetworkStore();
+      const result = await store.get();
+      expect(result).toStrictEqual(MOCK_STATE);
+    });
+
+    it('does not throw, and logs a debug message, if fetch is not okay', async () => {
+      const logDebugSpy = jest
+        .spyOn(log, 'debug')
+        .mockImplementation(() => undefined);
+
+      mockFixtureServerInterceptor().reply(400);
```

Comment:
Nit: The spacing here is a bit odd 🤔 Our docs recommend "arrange, act, assert" grouping in tests, i.e. separate the setup from the execution, and the execution from the assertions (and those from teardown, if there is any). More detail here: https://github.com/MetaMask/contributor-docs/blob/main/docs/testing/unit-testing.md#-highlight-the-exercise-phase

But instead there are newlines around the 2nd half of setup.

It'd be nice to have the standard spacing used for all tests, but it's especially odd here because my eye was drawn to this line as the "execution" phase, but that's not what it us.

Reply:
improved this and other tests in this file in a76aa9b8ca